core/core-impl/src/test/java/ai/timefold/solver/core/impl/solver/DefaultSolverTest.java

```diff
@@ -865,4 +964,74 @@ void solveWithMultipleChainedPlanningEntities() {
         assertThat(solution.getScore().isSolutionInitialized()).isTrue();
     }
 
+    public static class CorruptedEasyScoreCalculator implements EasyScoreCalculator<TestdataSolution, SimpleScore> {
```

Comment:
If only the method `calculatScore` is called during the solving process, then mocking it would make the test simpler.

Reply:
True. But since I need to go through the entire Solver hierarchy, I need an actual `class` to pass to the `SolverConfig`. An mock instance of the class will not be enough. Therefore I decided to do it like this.